### PR TITLE
[FIX] website: allow `load_menus_root` to force action

### PR DESCRIPTION
**Pull request by rdeodoo**
The website frontend apps menu list is not working when the user has a `Home Action` defined on his user.
The `Home Action` is meant to redirect to the defined action whenever that user is login in.
But since the backend menu links on the website have most of the time no `action` defined but just a `menu_id` defined, the `Home Action` will kick in and take over the redirection, the same way as if the user just type `/web` without any params.

To solve that, we simply force the `action` of those links (if they don't already have one).

This will make sure that the redirect is working as it should for users having a `Home Action` set.

--- Technical hints ---

There is multiple methods to get the list of menus in Odoo:
- `load_web_menus`: called by the web client rpc, calling `load_menus`. If a top/app menu has no action defined on it, it sets the first found action of their children menus to it. It returns the full (flat) list of menus, not only the top/app ones. This method is not ormcached but is calling an ormcached method and just doing some tiny work on the data.
- `load_menus_root`: called only by website backend template to add the app list on the website (in the frontend) to jump to the backend. It does not force the action if a menu has no action set on it. It returns only the top/app menus. This method is ormcached. Note that this method seems only used by the website module.
- `load_menus`: returns the full (flat) list of menus without a force action This method is ormcached.

Fixes https://github.com/odoo/odoo/issues/119971

### CODE REVIEW

**Comment by rdeodoo** [comment]
For:
- Odoo 16 forward port: The commit should be replaced by this one (already written) https://github.com/odoo/odoo/commit/99f6a18b39d87ac35024b9f0a2d3f408de50d1ef
  - `load_menus_root` is called directly through the XML backend view and not from the JS side anymore
  - The test need to be adapted to the new frontend overlay
- Odoo 14: There is no `load_web_menus` method, which is used in this fix, it's probably not worth spending time in this version to fix this bug.

File: addons/website/models/ir_ui_menu.py
Comment (rdeodoo) [inline addons/website/models/ir_ui_menu.py:13]: Note that this method is only used from `website` in our codebase.
In master, we may want to move it to `website` and always force the action.
I will open another PR once this is ported and merged in master as it will need to approval or the framework-py.

**Comment by rdeodoo** [comment]
@qsm-odoo Forgot to assign you last week, if you want to have a look :+1:

File: addons/website/models/ir_ui_menu.py
Comment (qsm-odoo) [inline addons/website/models/ir_ui_menu.py:15]: I always wonder: is not the "right" way to access using `.env.context` and `._context` is there for compatibility?

File: addons/website/models/ir_ui_menu.py
Comment (qsm-odoo) [inline addons/website/models/ir_ui_menu.py:20]: You already checked that in the above `if`

File: addons/website/tests/test_ui.py
Comment (qsm-odoo) [inline addons/website/tests/test_ui.py:317]: Nitpicking: Was confused by the assertion message, should not it be the opposite? "There is an action linked to the top menu which defeatss the purpose of the test" something like that? Otherwise you seem to say it is an error that the test relies on the fact the top menu has no action.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/models/ir_ui_menu.py
Comment (rdeodoo) [reply addons/website/models/ir_ui_menu.py]: Hm. I guess[ it is](https://github.com/odoo/odoo/blame/c30c1dac33da05a4064d5353215098b2a5b3c87d/odoo/models.py#L5112-L5115) even though it's still (more than) widely used.
I am not sure I knew about it or if I forgot, but I replaced it.

https://github.com/odoo/odoo/blob/c30c1dac33da05a4064d5353215098b2a5b3c87d/odoo/models.py#L5112-L5115

File: addons/website/models/ir_ui_menu.py
Comment (rdeodoo) [reply addons/website/models/ir_ui_menu.py]: Indeed, removed

File: addons/website/tests/test_ui.py
Comment (rdeodoo) [reply addons/website/tests/test_ui.py]: I think I see what you mean.
I removed the first part and moved it in some `()` so it does not looks like it's part of the problem (why this line failed) but looks like it explain why it should be the case.
```suggestion
        self.assertFalse(menu_root.action, 'The top menu should not have an action (or the test/tour will not test anything).')
```
(To keep the same sentence, I should've added it as a python comment above the line and omit the string in the `assert`.)

**Review by rdeodoo — COMMENTED** [review]
Changes: https://github.com/odoo/odoo/compare/feebcfbc18797cf93c270e89dd2e752c2514eb2f..70974b1a537b5469655c9b37b7bf50a5f2eb78d0

**Review by qsm-odoo — APPROVED** [review]
@robodoo r+

**Comment by qsm-odoo** [comment]
@rdeodoo I'll not ask to retry as the broken test in staging seems to be about your new test... race condition? :scream:

**Comment by rdeodoo** [comment]
Nah, it was because I forgot about enterprise.
I made this [change](https://github.com/odoo/odoo/compare/70974b1a537b5469655c9b37b7bf50a5f2eb78d0..b235fe18af6c457834d1e3c297462095e3fc639a), there is actually no bug on enterprise since we land on the app switcher and don't load the action from the frontend directly.

**Review by qsm-odoo — APPROVED** [review]
@robodoo r+

**Comment by qsm-odoo** [comment]
(CI failed, I rebuilt the sub-build)

**Pull request closed without merging**
